components/sts/org.wso2.carbon.identity.sts.passive/src/main/java/org/wso2/carbon/identity/sts/passive/processors/RequestProcessor.java

```diff
@@ -138,10 +152,30 @@ protected SAMLTokenIssuerConfig getSAMLTokenIssuerConfig(AxisService service, bo
             return null;
         }
 
+        tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```

Comment:
Check if tenant mode is working for Passive STS. Specifically do we set tenant domain to carbon context somewhere in the flow before reaching here ?